src/borg/patterns.py

```diff
@@ -340,6 +341,10 @@ def parse_pattern(pattern, fallback=FnmatchPattern, recurse_dir=True):
     """Read pattern from string and return an instance of the appropriate implementation class.
 
     """
+    # Potential optimization for fm, sh, and pf patterns: if recurse_dir is False and the pattern
+    # doesn't do any special matching, it could be converted to a PathFullPattern (we would also
+    # need to remove any trailing slashes from pf patterns, and exclude fm and sh patterns with
+    # trailing slashes from the optimization).
```

Comment:
why should we do any "optimization" or transformation?

we can rather keep it simple and just do what the user tells borg to do.

Reply:
The docs talk about how `pf` patterns perform better. It's implied that if you have a lot of patterns, it'd be better to use this style. It would be nice if users didn't have to worry about that, knowing the patterns they write will automatically optimized where possible.

Not a big deal though. Happy to add the context above to the comment, move it elsewhere, it remove it entirely.

Reply:
guess it is enough if the docs are clear about performance.

pf > pp > all else (speed wise).

should only matter if you have lots of patterns, so likely no issue for most users.